jobs/cloud/scheduler.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+package cloud
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/app"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+const (
+	SchedFreqMinutes = 2 // 1 day
+)
+
+type Scheduler struct {
+	App *app.App
+}
+
+func (m *CloudJobInterfaceImpl) MakeScheduler() model.Scheduler {
+	return &Scheduler{m.App}
+}
+
+func (scheduler *Scheduler) Name() string {
+	return JobName + "Scheduler"
+}
+
+func (scheduler *Scheduler) JobType() string {
+	return model.JOB_TYPE_CLOUD
+}
+
+func (scheduler *Scheduler) Enabled(cfg *model.Config) bool {
+	// If CloudUserLimit is non-zero we're in a cloud installation
+	return *cfg.ExperimentalSettings.CloudUserLimit > 0 && cfg.FeatureFlags.CloudEmailJobsEnabledMM29999
```

Comment:
AFAICT, there is no way to access the license from the Config struct, so I don't think so, but might be unaware of something.

Reply:
Is better to include the server in the scheduler struct instead of the app and use it to check the license too. If you check the line 45, you're using the server instead of the app. For this kind of things is better to depend on the server, instead of the app because the app is "request time object" kindof

Reply:
I didn't think to check the Scheduler. I've dropped the userlimit check in favour of a non-nil, cloud license + the feature flag here!